llvm/test/Feature/Repo/repo_pruning_dependents.py

```diff
@@ -0,0 +1,305 @@
+# These tests check the program repository pruning pass for the dependents.
+#
+# In the llvm-project-prepo project, the GO's digest could be generated either by the
+# RepoMetadataGenerator pass or by the RepoObjectWriter. When a new global object (GO) is
+# generated by an LLVM optimisation pass after the RepoMetadataGenerator pass, the digest
+# will be generated by the RepoObjectWriter. If a fragment has an external fixup
+# referencing this GO, we add the GO's compilation member address to the dependents.
+#
+# When a fragment is pruned, all its dependents need to be recorded in the repo.tickets
+# metadata to guarantee that the dependents (those GO generated by later optimizations)
+# are present in the final compilation record.
+#
+# ----  Test1 ----
+# In the test1, the digest of 'A' is given in the TicketNode (as if it were generated
+# by the RepoMetadataGenerator pass). The digests of `B` and `C` will be generated by the
+# RepoObjectWriter. Since Function 'A' is dependent on function 'B' and 'B' is dependent
+# on function 'C', both TicketNodes of 'B' and 'C' need to be added into the 'repo.tickets'
+# during the pruning.
+#
+# RUN: rm -f %t.db
+# Step 1: write a test1 IR code targeted on ELF.
+# RUN: python %s -t 1 -o %t_1.elf.ll
+# Step 2: generate a test1 IR code targeted on Repo.
+# RUN: env REPOFILE=%t.db clang -O0 -S -emit-llvm --target=x86_64-pc-linux-gnu-repo -x ir %t_1.elf.ll  -o %t_1.repo.ll
+# Step 3: remove the TicketNode metadata from the functions B and C.
+# RUN: python %s -f B C -i %t_1.repo.ll -o %t_1.ll
```

Comment:
I think this should use `%python` to pick up the python that lit is configured with, rather than one from the environment.

Reply:
Fixed, thanks.